src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -11,6 +11,55 @@
 from cryptography import utils
 
 
+class EllipticCurveField(object):
+    """
+    Field type of an EllipticCurve
+    """
+
+
+class EllipticCurvePrimeField(EllipticCurveField):
+    def __init__(self, p):
+        self._p = p
+
+    p = utils.read_only_property("_p")
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__) and self.p == other.p
```

Comment:
Our typical pattern for `__eq__` is to do 

```python
        if not isinstance(other, thisclass):
            return NotImplemented
```

as that allows the other class to potentially implement comparison. Could you make that change for these `__eq__` methods (and add a quick `!= object()` in the tests to exercise that code path)?